### PR TITLE
 Enable per pod PID namespace setting 

### PR DESCRIPTION
**Pull request by umohnani8**
If the pid namespace mode is set to POD, then the container's namespace
should be set to the namespace of the pod infra container.

To test this, make sure to add `FEATURE_GATES="${FEATURE_GATES},PodShareProcessNamespace=true"` to `hack/local-cluster-up.sh` - this enable the feature.

Signed-off-by: umohnani8 <umohnani@redhat.com>

### CODE REVIEW

**Comment by wking** [comment]
I think the CRI-O side of this needs to be a bit more nuanced than “ignore EnableSharedPIDNamespace if the kublet asks us to”.  Can we deprecate `EnableSharedPIDNamespace`?  What purpose does it serve now that CRI covers that idea?

#1280 is also in flight in this space, although it needs a rebase.  I'm fine with this PR going in first and don't expect significant conflicts; just something to be aware of.

**Comment by mrunalp** [comment]
I think that we can get rid of the flag as kube has a similar flag to enable/disable the feature.

**Comment by umohnani8** [comment]
Removed the `EnableSharedPIDNamespace` flag.
@mrunalp should I get rid of the namespace tests as the flag has been removed? Not sure how to test my changes as the pid namespace mode will be coming from kube.

**Comment by mrunalp** [comment]
@umohnani8 Yes, we should remove the existing tests that use that flag and add another integration test that sets pid namespace sharing to pod in the sandbox config. Take a look at my PR to update k8s that we merged. It has some similar changes to network namespace handling in the integration tests.

File: vendor.conf [hunk 1]
```diff
@@ -1,11 +1,11 @@
-k8s.io/kubernetes 7488d1c9210e60aef9ad49f07cb5d8a24152db88 https://github.com/kubernetes/kubernetes
+k8s.io/kubernetes fe0e80e8dafcdfde9778380f40346bfa75bad0f7 https://github.com/kubernetes/kubernetes
```
Comment (mrunalp) [inline vendor.conf]: Let us switch to the latest version in release-1.0 branch and also change to the same in the e2e tests https://github.com/kubernetes-incubator/cri-o/blob/master/contrib/test/integration/main.yml#L121

Comment (mrunalp) [reply vendor.conf]: https://github.com/kubernetes/kubernetes/commits/release-1.10

Comment (umohnani8) [reply vendor.conf]: Updated vendor and yaml to use the release 1.10 branch.
Added integration test.

**Comment by umohnani8** [comment]
Added test and update the main.yaml file.
Vendoring from last commit of release-1.10.

File: contrib/test/integration/main.yml [hunk 2]
```diff
@@ -118,7 +118,7 @@
       include: "build/kubernetes.yml"
       vars:
           force_clone: True
-          k8s_git_version: "master"
+          k8s_git_version: "release-1.10"
```
Comment (wking) [inline contrib/test/integration/main.yml]: Don't we intend to track k8s' master in our master branch?

File: cmd/crio/main.go [hunk 3]
```diff
@@ -131,9 +131,6 @@ func mergeConfig(config *server.Config, ctx *cli.Context) error {
 	if ctx.GlobalIsSet("pids-limit") {
 		config.PidsLimit = ctx.GlobalInt64("pids-limit")
 	}
-	if ctx.GlobalIsSet("enable-shared-pid-namespace") {
```
Comment (wking) [inline cmd/crio/main.go]: The man pages need updating too (e.g. [here][1]).

[1]: https://github.com/kubernetes-incubator/cri-o/blame/0b736bb43fb9f6ffa29b477be9e292bec3686ef4/docs/crio.8.md#L97

File: contrib/test/integration/main.yml (hunk 2 above)
Comment (mrunalp) [reply contrib/test/integration/main.yml]: @wking We will track release-1.10 in master till we make a branch for it next week. At that point we can switch back to master.

**Comment by rhatdan** [comment]
Since we have tests passing, should we merge now and get the man page fixes in a different PR?

File: cmd/crio/main.go (hunk 3 above)
Comment (umohnani8) [reply cmd/crio/main.go]: Updated the man page. Thanks for catching that!

**Comment by umohnani8** [comment]
@runcom should we run tests on this?

File: cmd/crio/main.go (hunk 3 above)
Comment (wking) [reply cmd/crio/main.go]: [Other man page][1] too ;).

[1]: https://github.com/kubernetes-incubator/cri-o/blame/0b736bb43fb9f6ffa29b477be9e292bec3686ef4/docs/crio.conf.5.md#L90

Comment (umohnani8) [reply cmd/crio/main.go]: Fixed. Removed it from helpers.bash as well. I think I got all of them now :)

**Comment by mrunalp** [comment]
restarted travis

**Pull request merged**